### PR TITLE
(#1572) Refactor site alert component

### LINKED ISSUE #1572: Make NCIDS usa-site-alert to be USWDS html
Site Alert came before we were patching. We should update site alert to match USWDS HTML. 

The main changes:
* `usa-site-alert--nci-standard` should go away, but we have to keep some of the styling that we did for handling the expand/collapse. This is basically the "default" variant, so we should not need a special class.
* `usa-site-alert--nci-info` and `usa-site-alert--nci-emergency` should just become the default USWDS modifiers. `usa-site-alert--info` and `usa-site-alert--emergency`. 
* `usa-site-alert--nci-slim` should go away and we just use `usa-site-alert--slim`
* We do wacky stuff for the contents of the alert:
   * ~~When we have expand/collapse we make the user add `usa-alert__nci-content` and `usa-alert__nci-header`. We should actually only add this from our Javascript.~~ The expand/collapse styles should be based on the aria-expanded value set in javascript.
   * USWDS allows the header to basically be a combination of lists and text blocks. We should follow their markup. 
     * When we are going to make it collapsible, it is only then do we take the header and other contents of the body and wrap it with the usa-alert__nci-content. (See Expand/Collapse Content Examples below)
     * On destroying the ncids component the DOM should be reset to the original HTML
* Strip the `<h3>` tags out of the html. Make sure that any H level works, as well as `<span>`.
* When there is no expand/collapse it should look like it does today with no JS firing. (But this should not be based on the styling of `usa-alert__nci-content` and `usa-alert__nci-header`!!!

#### Documentation site
- [x] Documentation site is missing variants and modifiers table and needs to follow component template structure:

   ```mdx
   variations:
     intro: |
       Choose from the following types of site alert components based on your page’s content design or organizational 
   needs.

       | Variant | Description |
       | ------- | ----------- |
       | Default | xxx |
       | Slim | xxx |

   modifications:
     intro: |
       Based on your page’s content design or organizational needs, you might need to modify the site alert further. The 
   following modifiers are available.

       | Modifier | Description |
       | -------- | ----------- |
       | Theme modifier | Info or emergency. Required, do not use any variant without a modifier. |
       
   ```

- [x] Missing auto init. See https://github.com/NCIOCPL/ncids/issues/1428
- [x] Update code snippets with breaking changes from ncids-css

#### `ncids-css`
- [x] Patch site alert so that the default variant keeps the expand/collapse css. This should remove `.usa-site-alert--nci-standard` classes everywhere.
- [x] ⚠️ **Breaking change**: Patch `.usa-site-alert--slim` to look like `.usa-sute-alert--nci-slim`. Replace this class in every template it is found. 
- [x] ⚠️ **Breaking change**: Patch `.usa-site-alert--info` to look like `.usa-site-alert--nci-info`. Replace this class in every template it is found. 
- [x] ⚠️ **Breaking change**: Patch `.usa-site-alert--emergency` to look like `.usa-site-alert--nci-emergency`. Replace this class in every template it is found. 
- [x] ⚠️ **Breaking change**: Remove `.usa-alert__nci-list` In template examples, replace with appropriate utility classes. We need to create a `.usa-list` modifier `.usa-list--nci-two-column`
- [x] Collapse/expand style should be on the `[aria-expanded=false/true]` css selector. Default should be collapsed to prevent CLS (unless the cookie says otherwise).
- [x] Remove NCIDS's site alert directory. We will now point to uswds.

> [!WARNING]  
> All breaking changes above should be backwards compatible for now. Deprecate the old classes with a note they will be removed in 3.0. That means the patch should contain the "bad" classes extending the good classes.

#### `ncids-twig`
- [x] Make a singular twig template for usa-site-alert that handles default and slim. See comments below.

#### `ncids-js`
- [x] Update any tests with breaking changes from ncids-css

#### `ncids-js` documentation
- [x] Create documentation for the site alert.

#### Storybook
- [x] Use ncids-twig with correct storybook structure.
- [x] Turn on NCIDS-js script for site alerts
- [x] Clean up storybook:

* 📁 Site alert
  * 📁 Default
    * Default
    * Dismissible
  * 📁 Modifiers
    * 📁 Emergency
      * Emergency default
      * Emergency slim
      * Dismissible emergency default
      * Dismissible emergency slim
    * 📁 Info
      * Info default
      * Info slim
      * Dismissible info default
      * Dismissible info slim
  * 📁 Variants
    * 📁 Slim
      * Slim
      * Slim Dismissible
  * 📁 Test cases
    * Emergency default with two column list
    * Info default with two column list

- [x] Create ticket for: #1717

   * Site alert
       * Default
       * Info
       * Emergency
       * Emergency list
       *  Emergency no header
       * Emergency no icon
       * Emergency slim

- [x] Update backstop scenarios. Remove tests for all breakpoints. Only add tests breakpoints used in site-alert.scss.

#### Example js site
- [x] Update example site HTML with breaking changes from ncids-css

#### Just text

```html
<div class="usa-alert__body">
  <span class="usa-alert__heading">Emergency alert message</span>
  <p class="usa-alert__text">
    Additional context and followup information including
    <a class="usa-link" href="javascript:void(0);">a link</a>.
  </p>
</div>
```

becomes

```html
<div class="usa-alert__body">
  <div class="usa-alert__nci-header">
    <span class="usa-alert__heading">Emergency alert message</span>
  </div>
  <div class="usa-alert__nci-content">
    <p class="usa-alert__text">
      Additional context and followup information including
      <a class="usa-link" href="javascript:void(0);">a link</a>.
    </p>
  </div>
</div>
```

#### Just a list

```html
<div class="usa-alert__body">
    <span class="usa-alert__heading">Emergency alert message</span>
    <ul class="usa-list">
      <li>
        The primary emergency message and
        <a class="usa-link" href="javascript:void(0);">a link</a> for
        supporting context.
      </li>
      <li>
        Another message,
        <a class="usa-link" href="javascript:void(0);">and another link</a>.
      </li>
      <li>A final emergency message.</li>
    </ul>
</div>
```

becomes

```html
<div class="usa-alert__body">
  <div class="usa-alert__nci-header">
    <span class="usa-alert__heading">Emergency alert message</span>
  </div>
  <div class="usa-alert__nci-content">  
    <ul class="usa-list">
      <li>
        The primary emergency message and
        <a class="usa-link" href="javascript:void(0);">a link</a> for
        supporting context.
      </li>
      <li>
        Another message,
        <a class="usa-link" href="javascript:void(0);">and another link</a>.
      </li>
      <li>A final emergency message.</li>
    </ul>
  </div>
</div>
```

#### Multi-things

```html
<div class="usa-alert__body">
  <span class="usa-alert__heading">Emergency alert message</span>
  <p class="usa-alert__text">
    Additional context and followup information including
    <a class="usa-link" href="javascript:void(0);">a link</a>.
  </p>
  <ul class="usa-list">
    <li>
      The primary emergency message and
      <a class="usa-link" href="javascript:void(0);">a link</a> for
      supporting context.
    </li>
    <li>
      Another message,
      <a class="usa-link" href="javascript:void(0);">and another link</a>.
    </li>
    <li>A final emergency message.</li>
  </ul>
  <p class="usa-alert__text">
    Additional context and followup information including
    <a class="usa-link" href="javascript:void(0);">a link</a>.
  </p>
</div>
```
becomes

```html
<div class="usa-alert__body">
  <div class="usa-alert__nci-header">
    <span class="usa-alert__heading">Emergency alert message</span>
  </div>
  <div class="usa-alert__nci-content">
    <p class="usa-alert__text">
      Additional context and followup information including
      <a class="usa-link" href="javascript:void(0);">a link</a>.
    </p>
    <ul class="usa-list">
      <li>
        The primary emergency message and
        <a class="usa-link" href="javascript:void(0);">a link</a> for
        supporting context.
      </li>
      <li>
        Another message,
        <a class="usa-link" href="javascript:void(0);">and another link</a>.
      </li>
      <li>A final emergency message.</li>
    </ul>
    <p class="usa-alert__text">
      Additional context and followup information including
      <a class="usa-link" href="javascript:void(0);">a link</a>.
    </p>
  </div>
</div>
```

### PR DESCRIPTION
**Pull request by olitharp-nci**
## Pull Request Details

- Add js to stories and reorganize storybook stories
- Add createAll method with `data-site-alert-closeable` attr
- Deprecate `NCISiteAlert`; use `USASiteAlert` instead (to be removed in 3.0.0)
- Add secure to close and collapse cookies
- Refactor html to more closely match USWDS
- Deprecate `usa-site-alert--nci-<modifier>` and `usa-alert__nci-list`  (to be removed in 3.0.0)
- Prevent CLS by defaulting to hidden content
- Create deprecated storybook tests
- Update scenarios
- Update type doc
- Update documentation site; replace code snippets with twig

Closes #1572

## Author PR Checklist
Items that the author of the PR is responsible for checking before submitted the PR.

### General:
- [x] I have reviewed the acceptance criteria defined in the ticket and ensured the work has been completed.
- [x] The commit message passes [all quality commit message standards](https://github.com/NCIOCPL/ncids/wiki/Creating-Quality-Commit-Messages).
- [x] Unit tests have been updated or created to reflect any javascript changes.
- [x] Storybook scenarios have been updated or created to reflect any html/css/js changes.

### Accessibility:
- [x] WCAG 2.1 Level AA requirements have been met.

### Development:
- [x] Any new or updated javascript code has 100% unit test coverage.
- [x] New or updated breakpoints have regression images.
- [x] Breaking changes have been thoroughly documented in the PR.

## Product Reviewer PR Checklist
Items the product team is responsible for reviewing.

### General:
- [x] There are no unexpected or unapproved regression image changes. 
- [x] Functionality of interactive elements meet the acceptance criteria.
- [x] The product is visually and functionally the same across the different browsers.

### Accessibility:
- [x] AxeDev Tools: there are no new or outstanding accessibility issues introduced in this PR.
- [x] Lighthouse: scores have not noticeably decreased during this PR.
- [x] Wave: there are no new errors or contrast errors introduced in this PR.

## Design Reviewer PR Checklist
Items the design team is responsible for reviewing.  

### General:
- [ ] New or updated features introduced in this PR are developed mobile-first.
- [ ] Breakpoint changes and regression images match those breakpoints.
- [ ] This PR has been tested in all supported browsers at all breakpoints.

## Developer Reviewer PR Checklist
Items the development team is responsible for reviewing.

### General:
- [ ] New code passes code quality standards set by industry standards.
- [ ] The expected Storybook stories have been added or updated for the new or updated feature.
- [ ] The expected unit tests have been added or updated for the new or updated feature.

### Accessibility:
- [ ] VoiceOver: Described content matches with what was expected.
- [ ] Keyboard navigation: new or updated features and content are navigable via the keyboard.

### CODE REVIEW

File: packages/ncids-js/src/components/usa-site-alert/usa-site-alert.component.ts [hunk 1]
```diff
@@ -0,0 +1,249 @@
...
+			);
+			siteAlerts.forEach((siteAlert) => {
+				const alert = siteAlert as HTMLElement;
+				const closeable =
+					alert.dataset.siteAlertClosable?.toLowerCase() === 'true';
+				USASiteAlert.create(alert as HTMLElement, { closeable });
+			});
+		});
+	}
+
```
Comment (bryanpizzillo) [inline packages/ncids-js/src/components/usa-site-alert/usa-site-alert.component.ts:184]: This should just call createAll.

**Review by bryanpizzillo — APPROVED** [review]
The code looks good.

File: packages/ncids-js/src/components/usa-site-alert/usa-site-alert.component.ts (hunk 1 above)
Comment (olitharp-nci) [reply packages/ncids-js/src/components/usa-site-alert/usa-site-alert.component.ts]: createAll is now called. Site alerts are up on the auto init testing page https://designsystem-dev.cancer.gov/pr-1708/example-site/example-pages/auto-init-example.html

**Review by andyvanavery31 — CHANGES_REQUESTED** [review]
Could you please add a vertical space between the Modifiers paragraph and the Emergency Modifier h3? 

image

**Review by andyvanavery31 — APPROVED** [review]
Approving this ticket with a plan to add a ticket to later fix the content of the documentation page.

**Pull request merged**